### PR TITLE
[MTDSA-2766] add Amend Charitable Giving Route

### PR DESCRIPTION
**Pull request by london-coder**
Add Charitable Giving amend (happy path) to router. Added resource, connector and service classes, and added to unit and integration tests for the version 2.0 Charitable Giving service.

### CODE REVIEW

File: it/router/resources/CharitableGivingResourceISpec.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import router.constants.Versions.VERSION_2
+
+class CharitableGivingResourceISpec extends ReleaseTwoIntegrationSpec {
+
+  val jsonRequest: JsObject = Json.obj("test" -> "json request")
+  val jsonResponse: JsObject = Json.obj("test" -> "json response")
+
+// NOT YET REQUIRED!!   At this time, only amend (PUT) is being addressed. But I leave this here for whom ever picks up retrieve
+/* "GET Charitable Giving annuals with release-2 enabled" should {
+
```
Comment (MarkAKelly) [inline it/router/resources/CharitableGivingResourceISpec.scala]: We should probably get rid of this if we don't need it yet.

File: it/support/FakeApplicationConfig.scala [hunk 2]
```diff
@@ -30,6 +30,8 @@ trait FakeApplicationConfig { _: WireMockConfig =>
     "microservice.services.auth.port" -> wiremockPort,
     "microservice.services.self-assessment-api.host" -> wiremockHost,
     "microservice.services.self-assessment-api.port" -> wiremockPort,
+    "microservice.services.mtd-charitable-giving-api.host" -> wiremockHost,
```
Comment (MarkAKelly) [inline it/support/FakeApplicationConfig.scala]: This should match the config from `app.conf` (`microservice.services.mtd-charitable-giving`).  This is why the tests are passing and didn't pick up the config change we made to the service name.

File: test/router/services/CharitableGivingServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    val requestBody = Json.obj("test" -> "body")
+
+    "return a HttpResponse" when {
+      "the request contains a version 1.0 header and release-2 config is disabled" in new Setup {
+        implicit val hc = HeaderCarrier(extraHeaders = Seq(ACCEPT -> "application/vnd.hmrc.1.0+json"))
+        val response = HttpResponse(200)
+        val release2EnabledConfig = Configuration("charitable-giving-version-2.enabled" -> false)
+        MockAppConfig.featureSwitch returns Some(release2EnabledConfig)
+
+        MockCharitableGivingConnector.put(request.uri, requestBody)
```
Comment (MarkAKelly) [inline test/router/services/CharitableGivingServiceSpec.scala]: Needs to be changed to go to `MockSelfAssessmentConnector` to go along with the code changes in the service.

File: test/router/services/CharitableGivingServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    )
+  }
+
+  implicit val request = FakeRequest()
+
+  "amend" should {
+    val requestBody = Json.obj("test" -> "body")
+
+    "return a HttpResponse" when {
+      "the request contains a version 1.0 header and release-2 config is disabled" in new Setup {
```
Comment (MarkAKelly) [inline test/router/services/CharitableGivingServiceSpec.scala]: The config is charitable giving version 2.

File: test/router/services/CharitableGivingServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        result shouldBe Right(response)
+      }
+
+      "the request contains a version 1.0 header and release-2 config is enabled" in new Setup {
+        implicit val hc = HeaderCarrier(extraHeaders = Seq(ACCEPT -> "application/vnd.hmrc.1.0+json"))
+        val response = HttpResponse(200)
+        val release2EnabledConfig = Configuration("charitable-giving-version-2.enabled" -> true)
+        MockAppConfig.featureSwitch returns Some(release2EnabledConfig)
+
+        MockCharitableGivingConnector.put(request.uri, requestBody)
```
Comment (MarkAKelly) [inline test/router/services/CharitableGivingServiceSpec.scala]: `MockSelfAssessmentConnector`

File: test/router/services/CharitableGivingServiceSpec.scala [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        result shouldBe Right(response)
+      }
+
+      "the request contains a version 2.0 header and release-2 config is disabled" in new Setup {
+        implicit val hc = HeaderCarrier(extraHeaders = Seq(ACCEPT -> "application/vnd.hmrc.2.0+json"))
+        val response = HttpResponse(200)
+        val release2EnabledConfig = Configuration("charitable-giving-version-2.enabled" -> false)
+        MockAppConfig.featureSwitch returns Some(release2EnabledConfig)
+
+        MockCharitableGivingConnector.put(request.uri, requestBody)
```
Comment (MarkAKelly) [inline test/router/services/CharitableGivingServiceSpec.scala]: `MockSelfAssessmentConnector`

**Comment by london-coder** [comment]
Hi @MarkAKelly I think I have addressed all you comments, can you please review again?

**Pull request merged**